sagemaker-spark/pyspark_mnist/pyspark_mnist_kmeans.ipynb

```diff
@@ -0,0 +1,285 @@
+    "express or implied. See the License for the specific language governing\n",
+    "permissions and limitations under the License."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# SageMaker PySpark K-Means Clustering MNIST Example\n",
+    "\n",
+    "1. [Introduction](#Introduction)\n",
+    "2. [Loading the Data](#Loading-the-Data)\n",
+    "3. [Training and Hosting a Model](#Training-and-Hosting-a-Model)\n",
+    "4. [Inference](#Inference)\n",
+    "5. [More on SageMaker Spark](#More-on-SageMaker-Spark)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "This notebook will show how to classify handwritten digits using the K-Means clustering algorithm through the SageMaker PySpark library. We will train on Amazon SageMaker using K-Means clustering on the MNIST dataset, host the trained model on Amazon SageMaker, and then make predictions against that hosted model.\n",
+    "\n",
+    "You can visit SageMaker Spark's Github repository at https://github.com/aws/sagemaker-spark for more about SageMaker Spark.\n",
+    "\n",
+    "We will train on Amazon SageMaker using the KMeans Clustering on the MNIST dataset, host the trained model on Amazon SageMaker, and then make predictions against that hosted model.\n",
+    "\n",
+    "First, we load the MNIST dataset into a Spark Dataframe, which dataset is available in LibSVM format at\n",
+    "\n",
+    "`s3://sagemaker-sample-data-[region]/spark/mnist/train/`\n",
+    "\n",
+    "where `[region]` is replaced with a supported AWS region, such as us-east-1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from pyspark import SparkContext, SparkConf\n",
+    "from pyspark.sql import SparkSession\n",
+    "\n",
+    "import sagemaker\n",
+    "import sagemaker_pyspark\n",
+    "from sagemaker import get_execution_role\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "\n",
+    "role = get_execution_role()\n",
+    "\n",
+    "# Configure Spark to use the SageMaker Spark dependency jars\n",
+    "jars = sagemaker_pyspark.classpath_jars()\n",
+    "\n",
+    "classpath = \":\".join(sagemaker_pyspark.classpath_jars())\n",
+    "\n",
+    "# See the SageMaker Spark Github repo under sagemaker-pyspark-sdk\n",
+    "# to learn how to connect to a remote EMR cluster running Spark from a Notebook Instance.\n",
+    "spark = SparkSession.builder.config(\"spark.driver.extraClassPath\", classpath)\\\n",
+    "    .master(\"local[*]\").getOrCreate()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# replace this with your own region, such as us-east-1\n",
+    "region = 'us-east-1'\n",
```

Comment:
Can we set region dynamically with `boto3.Session().region_name`?

Reply:
Done